core/semester_controller/test.py

```diff
@@ -0,0 +1,88 @@
+from django.test import TestCase
+from ta_scheduler.models import Semester
+from core.semester_controller.SemesterController import SemesterController
+from datetime import datetime
+
+
+class TestSemesterController(TestCase):
```

Comment:
Split this up into multiple classes. One for each method at least. (it's part of the course grading)